### DISCUSSION TITLE
Error: '(0 , crypto_1.diffieHellman) is not a function'

### DISCUSSION THREAD

**Discussion by daonv** [panva/jose#402] [Q&A] [answered]
Can anybody help show why the error happened when i running the below code? Thanks
Error: '(0 , crypto_1.diffieHellman) is not a function'
 
//the JWE format like this:    header.encryptedKey.iv.cipherText.tag
------------------------------------

 decryptJWE2 = async function decryptJWE2(header, encryptedKey, iv, cipherText, tag, privateKeyFile) {
  return new Promise(async (resolve, reject) => {
    try { 
        const privateKeyPem = fs.readFileSync(privateKeyFile, { encoding: "utf8" });
        const algorithm = 'ECDH-ES+A128KW';
        const ecPrivateKey = await jose2.importPKCS8(privateKeyPem, algorithm);
        const jwe = {
            ciphertext: cipherText,
            encrypted_key: encryptedKey,
            iv: iv,
            protected: header,
            tag: tag,
        };
      const {
            plaintext,
            protectedHeader
          } = await jose2.flattenedDecrypt(jwe, ecPrivateKey);
          console.log(protectedHeader);
          const decoder = new TextDecoder();
          console.log(decoder.decode(plaintext));
          console.log(decoder.decode(additionalAuthenticatedData));
      } catch(error) {
        console.log('error', error.message);
        reject(error);
      }
    });
}

**Comment by panva** [comment, marked as answer]
Node.js version not within the [supported](https://github.com/panva/jose/issues/262) version range.

**Reply by daonv** [reply]
it works perfect. Thanks you!
